Libraries/NativeAnimation/RCTNativeAnimatedModule.m

```diff
@@ -15,19 +15,20 @@
 @implementation RCTNativeAnimatedModule
 {
   RCTNativeAnimatedNodesManager *_nodesManager;
+
+  // Oparations called after views have been updated.
   NSMutableArray<AnimatedOperation> *_operations;
+  // Operations called before views have been updated.
+  NSMutableArray<AnimatedOperation> *_preOperations;
 }
 
 RCT_EXPORT_MODULE();
 
 - (void)invalidate
 {
   [_nodesManager stopAnimationLoop];
-}
-
-- (void)dealloc
-{
   [self.bridge.eventDispatcher removeDispatchObserver:self];
+  [self.bridge.uiManager removeUIManagerObserver:self];
 }
 
 - (dispatch_queue_t)methodQueue
```

Comment:
Actually the only reason we need to run on the UIManager queue now is to avoid having to lock `_operations` and `_preOperations.`. Added a comment to explain that.